src/strategies/offer_maker/market_making/Kandel/Kandel.sol

```diff
@@ -0,0 +1,265 @@
+// SPDX-License-Identifier:	BSD-2-Clause
+
+// Kandel.sol
+
+// Copyright (c) 2022 ADDMA. All rights reserved.
+
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+pragma solidity ^0.8.10;
+
+import {Direct, IMangrove, AbstractRouter, IERC20} from "mgv_src/strategies/offer_maker/abstract/Direct.sol";
+import {MgvLib, MgvStructs} from "mgv_src/MgvLib.sol";
+
+contract Kandel is Direct {
+  ///@notice number of offers managed by this strat
+  uint16 immutable NSLOTS;
+  ///@notice base of the market Kandel is making
+  IERC20 immutable BASE;
+  ///@notice quote of the market Kandel is making
+  IERC20 immutable QUOTE;
+  ///@notice quote distribution: `baseOfIndex[i]` is the amount of base tokens Kandel must give or want at index i
+  uint[] baseOfIndex;
```

Comment:
consider: IMHO At works better han Of for most of these names, e.g., baseAtIndex

Reply:
It's functional style naming scheme f of i is f(i)